### ISSUE TITLE
Allow non default txt's in scripts/ and allow txt's and png's from sub folders to be uploaded to workshop.

### ISSUE THREAD

**Issue by MagentaIris** [Facepunch/garrysmod-requests#2426]
This has been bothering me for a long time.

I can't upload .txt's or .png's to workshop if they're inside the Sound folder, even though I can READ txt's and png's from there.

They are bundled together for clarity and ease of use of the custom music system for my gamemode.

I had to change this for the workshop version due to this really stupid arbitrary limit.

Alongside this, you allow scripts/vehicles/X.txt to be uploaded, but not scripts/X.txt even though sound scripts are often added via scripts/X.txt, meaning if I wanted to add sound scripts to a workshop file I'd either have to hide them as .lua's or put them into scripts/vehicles/X.txt

All of this is absolutely stupid, If you're scared of people overwriting default script names or files, make a white list of default script names and files.

**TL;DR**

Let use upload txt's and png's in sub folders that aren't just scripts/vehicles and materials/, because some of us actually like having files in the same location as a system created in-engine warrants it. It just makes uploading to workshop more cumbersome and shitty due to its arbitrary limitations of where I can read files from.

**Comment by garryspins** [comment]
Why would you want to put pngs in a sound folder?

**Comment by MagentaIris** [comment]
> Why would you want to put pngs in a sound folder?

Like I said in the original post, for Clarity and Ease of Use.

Music Packs for my gamemode are created by using only a single folder in sound/TemporalAssassin/Music/PACK_NAME_HERE/[FILES]

The files consist of .txt's to identify titles and other information of mp3s or oggs, and a single png is used to define the Cover Art that shows up on screen when music from this pack is played.

It's all in the same folder so its easier to create music packs, especially for newer people.

**Comment by robotboy655** [comment]
You can already ship `.vcd` files in `scenes/`. 

You can ship `.png` files in `materials/` folder. You can ship `.txt` files in `data_static/` folder.

`scripts/` folder will not have `.txt` files allowed, as that allows you to overwrite core game files that can lead to unwanted, or even dangerous issues.

Soundscripts are not automatically loaded from `scripts/` folder anyway, they use a manifest file, which cannot be allowed to overwritten by addons for what should be obvious reasons - addons could simply make all base game soundscripts not work, have conflicts between each other, etc. Perhaps a Lua function to load a .txt file as a sountscript file would be a more appropriate request, similarly to how other functions such as `PrecacheSentenceFile`, `game.AddParticles` already work.

**Comment by garryspins** [comment]
Makes sense to me actually, folders exist for a reason, whats stopping you from defining stuff in code? I get you're trying to do something fancy but look at other solutions.

**Comment by MagentaIris** [comment]
> Makes sense to me actually, folders exist for a reason, whats stopping you from defining stuff in code? I get you're trying to do something fancy but look at other solutions.

Mostly because having to redo a shit ton of definitions and code just because workshop cries sucks ass.

Along side that a lot of stuff was already made for the current system I've created, so I'm having to spend even MORE time "Fixing" this when it wouldn't need to be fixed if workshop wasn't so anal about file locations.
